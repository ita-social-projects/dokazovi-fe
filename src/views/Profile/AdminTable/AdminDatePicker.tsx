import React from 'react';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import { parsDate } from 'utilities/parsDate';
import { DatePicker } from '@material-ui/pickers';
import { useTranslation } from 'react-i18next';
import { Box } from '@material-ui/core';
import { langTokens } from '../../../locales/localizationInit';
import { useStyles } from './styles/AdminDatePicker.styles';

type VerticalType = number | 'bottom' | 'top' | 'center';
type HorizontalType = number | 'center' | 'left' | 'right';

const MenuProps = {
  disableScrollLock: true,
  anchorOrigin: {
    vertical: 'bottom' as VerticalType,
    horizontal: 'left' as HorizontalType,
  },
  transformOrigin: {
    vertical: 'top' as VerticalType,
    horizontal: 'left' as HorizontalType,
  },

  getContentAnchorEl: null,
};

interface IMaterialsDate {
  start: string | undefined;
  end: string | undefined;
  setChanges: (payload: any) => void;
}

export const AdminDatePicker: React.FC<IMaterialsDate> = ({
  start,
  end,
  setChanges,
}) => {
  const classes = useStyles();
  const { t } = useTranslation();

  return (
    <div>
      <FormControl>
        <Select
          multiple
          value={[]}
          disableUnderline
          displayEmpty
          renderValue={() => (
            <div>
              {t(langTokens.admin.from)} {parsDate(start)}{' '}
              {t(langTokens.admin.to)} {parsDate(end)}
            </div>
          )}
          MenuProps={MenuProps}
        >
          <Box display="flex" className={classes.pickersWrapper}>
            <DatePicker
              autoOk
              orientation="portrait"
              variant="static"
              openTo="date"
              value={start}
              onChange={(value) =>
                setChanges({
                  option: 'start',
                  date: value && value.toISOString(),
                })
              }
              format="dd-mm-yyyy"
              maxDate={end}
            />
            <DatePicker
              autoOk
              orientation="portrait"
              variant="static"
              openTo="date"
              value={end}
              onChange={(value) =>
                setChanges({
                  option: 'end',
                  date: value && value.toISOString(),
                })
              }
              format="dd-mm-yyyy"
              minDate={start}
            />
          </Box>
        </Select>
      </FormControl>
    </div>
  );
};
