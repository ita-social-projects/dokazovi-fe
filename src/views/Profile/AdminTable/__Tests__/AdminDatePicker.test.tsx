import React from 'react';
import { render, screen } from '@testing-library/react';
import { AdminDatePicker } from '../AdminDatePicker';

const mockOnSetChanged = jest.fn();

test('component renders properly with all props', () => {
  render(
    <AdminDatePicker
      start="01-01-2000"
      end="01-02-2000"
      setChanges={mockOnSetChanged}
    />,
  );

  expect(screen.getByTestId('select')).toBeInTheDocument();
});
