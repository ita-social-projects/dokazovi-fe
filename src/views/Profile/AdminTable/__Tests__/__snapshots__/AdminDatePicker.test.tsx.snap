// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders properly with all props 1`] = `
<div
  class="MuiFormControl-root"
  data-testid="select"
>
  <div
    class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
  >
    <div
      aria-haspopup="listbox"
      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      role="button"
      tabindex="0"
    >
      <div>
        з
         
        31.12.1999
         
        по
         
        01.01.2000
      </div>
    </div>
    <input
      aria-hidden="true"
      class="MuiSelect-nativeInput"
      tabindex="-1"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
